[MQ-94] Tests that anon data toggle can be toggled

michelleshull opened the pull request.

michelleshull commented:
This is a super simple test that only tests if the toggle toggles, it doesn't check to see if the db table is changed, which is the only way I can find that would assert that sending is off or on, and is out of the purview of an acceptance test.

JanJakes commented on tests/acceptance/AdvancedSettingsAnonDataToggleCest.php:
```diff
@@ -0,0 +1,19 @@
+
+class AdvancedSettingsAnonDataToggleCest {
+  function toggleAnonDataSetting(\AcceptanceTester $I) {
+    $I->wantTo('Confirm anon data settings can be toggled');
```
Could we use the full word? I had no idea what `anon` is until I opened full advanced.html. Also the class could be just `AdvancedSettingsCest` maybe? To possibly contain multiple test in future.

michelleshull replied:
Sure thing. I'll probably create new classes for the other advanced settings tests, so I changed the names to include anonymous instead of anon.

JanJakes commented on tests/acceptance/AdvancedSettingsAnonDataToggleCest.php:
```diff
@@ -0,0 +1,19 @@
+    $I->selectOption($noAnonData, 'No');
+    $I->seeOptionIsSelected($noAnonData, 'No');
+    $I->selectOption($yesAnonData, 'Yes');
+    $I->seeOptionIsSelected($yesAnonData, 'Yes');
```
I think one way to check if this was saved in the DB is to refresh the page.

michelleshull replied:
Hahah, smart and should have been obvious. Done. I saved between toggles, as well.

JanJakes commented on tests/acceptance/AdvancedSettingsAnonDataToggleCest.php:
```diff
@@ -2,18 +2,24 @@
+    $I->selectOption($yesAnonymousData, 'Yes');
+    $I->click($submitButton);
+    $I->waitForElementClickable($submitButton);
+    $I->seeOptionIsSelected($yesAnonymousData, 'Yes');
```
Soo... I meant more something like this:
```
$I->selectOption($noAnonymousData, 'No');
$I->click($submitButton);
$I->waitForText('Settings saved');
$I->reloadPage();
$I->seeOptionIsSelected($noAnonymousData, 'No');
```

Meaning that I see the option correctly selected after change & save & reload. I guess no need to reload between testing the two options.

michelleshull replied:
Saving refreshes the page, so I cut out the reload all together.

michelleshull commented:
I changed the name to AdvancedSettingsCest, so this class can be expanded as I add additional tests for this page, without making an overly large PR. Added a test for the bounce email, and utilized the System Info output to confirm the change.

JanJakes commented on views/settings/advanced.html:
```diff
@@ -9,7 +9,7 @@
           <%= __('Your bounced emails will be sent to this address.') %>
           <a href="http://beta.docs.mailpoet.com/article/180-how-bounce-management-works-in-mailpoet-3"
-             target="_blank"
+             target="_blank" data-automation-id="bounce-kb-link"
```
This is not used anywhere so I guess it can be dropped from this pull request.

JanJakes commented on tests/acceptance/AdvancedSettingsCest.php:
```diff
@@ -0,0 +1,48 @@
+    $I->click($submitButton);
+    $I->waitForText($successMessage);
+    $I->seeOptionIsSelected($yesAnonymousData, 'Yes');
+  }
```
There should be an empty line between methods. We need to add linters for this though.

michelleshull commented:
Thanks for the great feedback and making me work for this one. :)
